src/OSPSuite.Presentation/Presenters/Charts/DataBrowserPresenter.cs

```diff
@@ -58,32 +61,27 @@ public interface IDataBrowserPresenter : IPresenter<IDataBrowserView>, IPresente
       /// <summary>
       ///    Update the used/not used state for a list of columns
       /// </summary>
-      /// <param name="usedDataColumns">The list of columns</param>
-      /// <param name="usedState">true if the columns should be used, otherwise false</param>
-      void SetUsedState(IReadOnlyList<DataColumn> usedDataColumns, bool usedState);
-
-      /// <summary>
-      ///    Returns the descendant data repoository column ids. Distinct from <see cref="SelectedDataColumns" /> because
-      ///    this will return all ancestor data from a group row in the data grid
-      /// </summary>
-      /// <returns>all selected rows, either by direct selection, or by parent relationship</returns>
-      IReadOnlyList<DataColumn> SelectedDescendentDataRepositoryColumns { get; }
+      /// <param name="dataColumnDTOs">The list of columns</param>
+      /// <param name="used">true if the columns should be used, otherwise false</param>
+      void SetUsedState(IReadOnlyList<DataColumnDTO> dataColumnDTOs, bool used);
 
       IReadOnlyList<DataColumn> SelectedDataColumns { get; }
+
+      void UsedChangedFor(DataColumnDTO dto, bool used);
+      void UpdateUsedForSelection(bool used);
    }
 
    internal class DataBrowserPresenter : PresenterWithColumnSettings<IDataBrowserView, IDataBrowserPresenter>, IDataBrowserPresenter
    {
-      private readonly DataColumnsDTO _dataColumnsDTO;
-      private readonly Cache<string, DataColumn> _dataColumnCache;
+      private readonly NotifyCache<DataColumn, DataColumnDTO> _dataColumnDTOCache;
```

Comment:
We don't use a DataTable with weird strings as DTO. Instead, we have as usual a list of DTO Objects and all falls into place